java/java.disco/src/org/netbeans/modules/java/disco/Client.java

```diff
@@ -68,28 +68,31 @@ private synchronized DiscoClient getDisco() {
     private Client() {
     }
 
-    public synchronized final List<MajorVersion> getAllMajorVersions() {
+    /**
+     * Returns all major versions which are still maintained (excludes EA releases).
+     */
+    public synchronized final List<MajorVersion> getAllMaintainedMajorVersions() {
         if (majorVersions == null) {
-            majorVersions = Collections.unmodifiableList(
-                    getDisco().getAllMajorVersions(false).stream()
-                            .filter(majorVersion -> majorVersion.isMaintained())
-                            .filter(majorVersion -> majorVersion.isEarlyAccessOnly() != Boolean.TRUE)
-                            .collect(Collectors.toList()));
+            majorVersions = getDisco().getAllMajorVersions(
+                    Optional.of(true),   // maintained
+                    Optional.of(false),  // EA
+                    Optional.of(true),   // GA
+                    Optional.of(false)); // build
         }
         return majorVersions;
     }
 
-    public synchronized MajorVersion getLatestLts(boolean b) {
-        return getDisco().getLatestLts(b);
+    public synchronized MajorVersion getLatestLts(boolean includeEA) {
+        return getDisco().getLatestLts(includeEA);
     }
 
-    public synchronized MajorVersion getLatestSts(boolean b) {
-        return getDisco().getLatestSts(b);
+    public synchronized MajorVersion getLatestSts(boolean includeEA) {
+        return getDisco().getLatestSts(includeEA);
     }
 
     public synchronized List<Pkg> getPkgs(final Distribution distribution, final VersionNumber versionNumber, final Latest latest, final OperatingSystem operatingSystem,
             final Architecture architecture, final ArchiveType archiveType, final PackageType packageType,
-            final Boolean javafxBundled) {
+            final boolean ea, final boolean javafxBundled) {
```

Comment:
Disco does actually use `Boolean` with `null` for unspecified here! :grimacing:

Reply:
Agreed, not public so can be changed later - just mentioned because it was a direct mirror of upstream as far as I recall, and I thought it was weird too!